integration-tests/src/tests/client/features/increase_storage_compute_cost.rs

```diff
@@ -0,0 +1,369 @@
+//! Tests to verify the compute cost limit for storage operations are increased
+//! for protocol version 61. See `core/primitives/res/runtime_configs/61.yaml`
+//! for the exact changes.
+//!
+//! We test `storage_write` and `storage_remove` because they can easily be
+//! tested through existing methods `insert_strings` and `delete_strings` in
+//! `near_test_contracts::rs_contract()`. This doesn't cover all the changed
+//! costs individually but if it works for two parameters we can be reasonably
+//! confident that it works for the others as well.
+//! We also test unaffected cases to make sure compute costs only affect
+//! parameters they should.
+
+use assert_matches::assert_matches;
+use near_chain::ChainGenesis;
+use near_chain_configs::Genesis;
+use near_client::test_utils::TestEnv;
+use near_client::ProcessTxResponse;
+use near_crypto::{InMemorySigner, KeyType};
+use near_epoch_manager::shard_tracker::TrackedConfig;
+use near_primitives::config::ActionCosts;
+use near_primitives::runtime::config_store::RuntimeConfigStore;
+use near_primitives::sharding::ShardChunk;
+use near_primitives::transaction::{
+    Action, DeployContractAction, FunctionCallAction, SignedTransaction,
+};
+use near_primitives::types::AccountId;
+use near_store::test_utils::create_test_store;
+use nearcore::config::GenesisExt;
+use nearcore::NightshadeRuntime;
+use node_runtime::config::RuntimeConfig;
+use std::path::Path;
+
+/// Test that `storage_write` compute limit is respected in new version.
+#[test]
+fn test_storage_write() {
+    //  `insert_strings(from: u64, to: u64)` makes (`to` - `from`) `storage_write` calls.
+    let method_name = "insert_strings".to_owned();
+    let num_writes = 100u64;
+    let method_args: Vec<u8> =
+        0u64.to_le_bytes().into_iter().chain(num_writes.to_le_bytes()).collect();
+    let num_transactions = 200;
+    let uses_storage = true;
+    let fails = false;
+    assert_compute_limit_reached(method_name, method_args, num_transactions, uses_storage, fails);
+}
+
+/// Test that `storage_remove` compute limit is respected in new version.
+#[test]
+fn test_storage_remove() {
```

Comment:
So it doesn't actually remove anything but costs are still charged, right?

Reply:
yes, exactly, the gas is always charged regardless of the result